dashboard/src/components/features/projects/Branch/ManageBranchModal.tsx

```diff
@@ -0,0 +1,90 @@
+import { DialogContent, DialogDescription, DialogFooter, DialogHeader, DialogTitle } from "@/components/ui/dialog"
+import { ProjectData } from "../Projects"
+import { Button } from "@/components/ui/button"
+import { KeyedMutator } from "swr"
+import { Checkbox } from "@/components/ui/checkbox"
+import { useState } from "react"
+import { useFrappeDeleteDoc } from "frappe-react-sdk"
+import { CommitProjectBranch } from "@/types/commit/CommitProjectBranch"
+import { toast } from "@/components/ui/use-toast"
+
+export interface ManageBranchModalProps {
+    branches: CommitProjectBranch[]
+    mutate: KeyedMutator<{ message: ProjectData[]; }>
+    setOpenManageModal: React.Dispatch<React.SetStateAction<boolean>>
+}
+
+
+const ManageBranchModal = ({ branches, mutate, setOpenManageModal }: ManageBranchModalProps) => {
+
+
+    const [branchList, setBranchList] = useState<string[]>([])
+
+
+    const { deleteDoc, reset } = useFrappeDeleteDoc()
+
+    const handleDelete = (branchList: string[]) => {
+        branchList.map((branchListItem) => {
+            deleteDoc("Commit Project Branch", branchListItem)
+                .then(() => {
+                    setBranchList([])
+                    mutate()
+                    reset()
+                    setOpenManageModal(false)
+
+                }).then(() => toast({
+                    description: "Branch Deleted",
+                    variant: "destructive",
+                }))
+        })
+    }
+
+    return (
+        <DialogContent className="sm:max-w-[600px] sm:max-h-[800px] overflow-y-scroll">
+            <DialogHeader>
+                <DialogTitle>Manage Branches</DialogTitle>
+                <DialogDescription>
+                    Select Branches to delete.
```

Comment:
Change this text as this modal will have multiple use Case like delete fetch repo